lib/rubygems/specification.rb

```diff
@@ -2505,6 +2506,28 @@ def traverse trail = [], &block
     end
   end
 
+  def self.traverse spec, &block # :nodoc:
+    _traverse spec, Gem::List.new(spec), &block
+  end
+
+  ##
+  # This method is for traversing spec dependencies.  Don't use this, it is
+  # super private. I am super serious!
+
+  def self._traverse spec, trail, &block # :nodoc:
+    spec.dependencies.each do |dep|
+      next unless dep.runtime?
+      dep.to_specs.each do |dep_spec|
+        stack = Gem::List.new(dep_spec, trail)
+        block[dep_spec, stack]
+        spec_name = dep_spec.name
+        _traverse(dep_spec, stack, &block) unless
```

Comment:
I may have missed something, but I think this should be `traverse` rather than `_traverse` as the former is the only one (I can see) with a 3 element signature.